src/compiler_internals.md

```diff
@@ -0,0 +1,37 @@
+# Compiler Internals
+
+Alan's compiler is a multi-stage compiler written mostly in Typescript at the moment. There are two primary targets, Javascript and the Alan Runtime's Alan Graphcode (AGC) format, with a branching path involving two intermediate representations: Alan-- (AMM) and Alan Graphcode Assembler (AGA).
```

Comment:
We should imply that the AGA + AGC branch is the main path and JS, while completely supported, is a secondary transpilation target (and link to that section of the docs). This is mostly to avoid the confusion some folks had when we launched about how they thought Alan only compiled to JS.

Reply:
That makes sense. Need to think a bit on how to rephrase this.